tsfc/tsfc_to_loopy.py

```diff
@@ -416,19 +438,34 @@ def subscr(name, indices):
 
     # }}}
 
-    data = [
+    global_temps = [
         lp.TemporaryVariable(
             name, shape=lp.auto, initializer=val,
             scope=lp.temp_var_scope.GLOBAL,
             read_only=True)
-        for name, val in six.itervalues(ctx.literal_to_name_and_array)] + ["..."]
+        for name, val in six.itervalues(ctx.literal_to_name_and_array)]
+    cse_temps = [
+        lp.TemporaryVariable(
+            name, dtype=np.float64, shape=lp.auto, base_indices=lp.auto,
+            scope=lp.temp_var_scope.PRIVATE) for name in ctx.cse_names]
+    data = global_temps + cse_temps + ["..."]
 
     knl = lp.make_kernel(
         [domain],
         instructions + ctx.subst_rules,
         data,
         name=kernel_name)
 
+    # FIXME: Dealing with the Island problem
+    # Commenting(and not deleting) it as it maybe used for temporary
+    # compilation. A long term solution is in process
+    '''
+    for insn_id, cse_assignment in enumerate(ctx.assignments):
+        var_name, free_indices, _ = cse_assignment
+        for iname in free_indices:
+            knl = lp.duplicate_inames(knl, iname, "writes:"+var_name)
+    '''
```

Comment:
How about a user-exposed flag until the long-term solution arrives? (So that the commenting is not necessary?)